### ISSUE TITLE
Run `go mod tidy` in Go projects

### ISSUE THREAD

**Issue by TheRedSpy15** [dependabot/feedback#215]
This would be awesome to have as a Go developer

**Comment by feelepxyz** [comment]
Pinging @hmarr who's currently rewriting our go mod support!

**Comment by hmarr** [comment]
@TheRedSpy15 - I'm with you, I'd love to add `go mod tidy` support. It's a bit tricky currently, as we'd need to clone the entire repo (`go mod tidy` looks at your source code to determine what should be present in the `go.mod`), and Dependabot is currently only set up to fetch individual dependency files.

However, (optionally) cloning the whole repo is something we'd like to do at some point, as it'd also let us handle major version updates for Go modules, and help us with vendoring support.

**Comment by RXminuS** [comment]
Actually, maybe there's a more overarching need to run any command as part of generating the PR? I would love to run a certain Makefile command if a submodule updates for instance (Re-generate the GRPC code when the submodule with .proto files is updated for instnace). Seems that his would also fit within that

**Comment by hmarr** [comment]
Sounds like a great use case for [GitHub Actions](https://blog.github.com/2018-10-17-action-demos/). You could run an action to regenerate the GRPC code and push a new commit to any PR that touches a .proto file, or run `go mod tidy` on PRs tagged with `dependencies`.

**Comment by paultyng** [comment]
`go mod vendor` would be another command that would be useful, but it looks like similar problem to tidy, you don't have the whole repo. Maybe something we can do with an action as well in the meantime.

**Comment by hmarr** [comment]
Vendoring is on my list, too. Once we're able to make larger changes across the repo, we should be able to handle vendoring as well as `go mod tidy`.

**Comment by paultyng** [comment]
I have about 80+ repositories in this org alone I'd like to use this on: https://github.com/terraform-providers but we still vendor until we get a module proxy setup.

**Comment by sylr** [comment]
Also really keen to have `go mod vendor`

**Comment by miyajan** [comment]
Are there any updates on this issue? We hope that `go mod tidy` is executed for Go modules update.

**Comment by hmarr** [comment]
We've been busy with the acquisition recently so haven't made progress on this yet. I'm hoping we'll be able to build out better Go support over the next few months though.

**Comment by racevedoo** [comment]
`go mod tidy` would be great!

**Comment by MrJoy** [comment]
I'm also going to chime in on vendoring support.  Ideally, I'd prefer it if both go modules and `dep` were supported, since my org has a couple repos where we rely on tools that aren't module-friendly yet, and thus won't be migrating those repos over for a while.

**Comment by fallion** [comment]
Hi, 
Any update on this?

**Comment by greysteil** [comment]
Afraid not - the Dependabot team is still tiny and working to get to 100% rollout for automated security fixes before tackling additional feature work.

**Comment by fallion** [comment]
Is there a way to help?

**Comment by greysteil** [comment]
I don't think so - as @hmarr says above this would be a pretty major change as we'd need to start cloning your repo rather than just fetching your dependency files, which in turn would need a different infrastructure setup to support it.

**Comment by dentarg** [comment]
I think this issue should be kept open.

**Comment by crazy-max** [comment]
My current workaround is using [this workflow](https://github.com/crazy-max/diun/blob/master/.github/workflows/gosum.yml) to [remove `go.sum` and exec a `go mod tidy`](https://github.com/crazy-max/diun/blob/master/.github/workflows/gosum.yml#L26-L27) and push changes.

**Comment by sue445** [comment]
As another solution, I made https://github.com/marketplace/actions/go-mod-tidy-pr
